### DISCUSSION TITLE
Shared `_internal` folder cross multiple pyinstaller packages

### DISCUSSION THREAD

**Discussion by tigerinus** [pyinstaller/pyinstaller#8208] [Help] [answered]
We have several AI apps built on top of CUDA+pytorch. Each app packaged by pyinstaller is around 2~3GB in size.

Would it be possible to build multiple apps together, say under the same virtualenv, so it generates a `dist` folder with shared `_internal` folder?

**Comment by rokm** [comment, marked as answer]
You can create a .spec file that uses a shared `COLLECT`: https://www.zacoding.com/en/post/pyinstaller-create-multiple-executables/

Basically, take the contents from your multiple applications' spec files, rename the variables so that they are unique, and then use a single COLLECT with EXE instances, and all `datas` and `binaries` from all `Analysis` instances.

**Reply by tigerinus** [reply]
Great.

This should be part of official doc.

**Reply by rokm** [reply]
> Great.
> 
> This should be part of official doc.

https://pyinstaller.org/en/stable/spec-files.html#multipackage-bundles
